ctags/main/main.c

```diff
@@ -0,0 +1,277 @@
+# endif
+#else
+# ifdef HAVE_TIMES
+#  ifdef HAVE_SYS_TIMES_H
+#   include <sys/times.h>
+#  endif
+# endif
+#endif
+
+/*  To provide directory searching for recursion feature.
+ */
+
+#ifdef HAVE_DIRENT_H
+# ifdef HAVE_SYS_TYPES_H
+#  include <sys/types.h>  /* required by dirent.h */
+# endif
+# include <dirent.h>  /* to declare opendir() */
+# undef boolean
+#endif
+#ifdef HAVE_DIRECT_H
+# include <direct.h>  /* to _getcwd() */
+#endif
+#ifdef HAVE_DIR_H
+# include <dir.h>  /* to declare findfirst() and findnext */
+#endif
+#ifdef HAVE_IO_H
+# include <io.h>  /* to declare _findfirst() */
+#endif
+
+#include <string.h>
+
+#include "debug.h"
+#include "entry.h"
+#include "keyword.h"
+#include "main.h"
+#include "options.h"
+#include "read.h"
+#include "routines.h"
+
+/*
+*   MACROS
+*/
+#define plural(value)           (((unsigned long)(value) == 1L) ? "" : "s")
+
+/*
+*   DATA DEFINITIONS
+*/
+static struct { long files, lines, bytes; } Totals = { 0, 0, 0 };
+
+/*
+*   FUNCTION DEFINITIONS
+*/
+
+extern void addTotals (
+		const unsigned int files, const long unsigned int lines,
+		const long unsigned int bytes)
+{
+	Totals.files += files;
+	Totals.lines += lines;
+	Totals.bytes += bytes;
+}
+
+extern boolean isDestinationStdout (void)
+{
+	boolean toStdout = FALSE;
+
+	if (Option.xref  ||  Option.filter  ||
+		(Option.tagFileName != NULL  &&  (strcmp (Option.tagFileName, "-") == 0
+						  || strcmp (Option.tagFileName, "/dev/stdout") == 0
+		)))
+		toStdout = TRUE;
+	return toStdout;
+}
+
+
+# if defined (WIN32)
+
+static boolean createTagsForMatchingEntries (char *const pattern)
+{
+	boolean resize = FALSE;
+	const size_t dirLength = baseFilename (pattern) - (char *) pattern;
+	vString *const filePath = vStringNew ();
+#if defined (HAVE_FINDFIRST)
+	struct ffblk fileInfo;
+	int result = findfirst (pattern, &fileInfo, FA_DIREC);
+	while (result == 0)
+	{
+		const char *const entryName = fileInfo.ff_name;
+
+		/*  We must not recurse into the directories "." or "..".
+		 */
+		if (strcmp (entryName, ".") != 0  &&  strcmp (entryName, "..") != 0)
+		{
+			vStringNCopyS (filePath, pattern, dirLength);
+			vStringCatS (filePath, entryName);
+			resize |= createTagsForEntry (vStringValue (filePath));
+		}
+		result = findnext (&fileInfo);
+	}
+#elif defined (HAVE__FINDFIRST)
+	struct _finddata_t fileInfo;
+	long hFile = _findfirst (pattern, &fileInfo);
+
+	if (hFile != -1L)
+	{
+		do
+		{
+			const char *const entryName = fileInfo.name;
+
+			/*  We must not recurse into the directories "." or "..".
+			 */
+			if (strcmp (entryName, ".") != 0  &&  strcmp (entryName, "..") != 0)
+			{
+				vStringNCopyS (filePath, pattern, dirLength);
+				vStringCatS (filePath, entryName);
+				resize |= createTagsForEntry (vStringValue (filePath));
+			}
+		} while (_findnext (hFile, &fileInfo) == 0);
+		_findclose (hFile);
+	}
+#endif
+
+	vStringDelete (filePath);
+	return resize;
+}
+
+#endif
+
+
+static boolean recurseIntoDirectory (const char *const dirName)
```

Comment:
Should I #if0 it or keep it the way it is? There will always be some dead code and the few lines here won't help much I think.